### PR TITLE
Add Code to validator struct.

### PR DESCRIPTION
**Pull request by sugarraysam**
This is the first PR for this ticket: https://issues.redhat.com/browse/MTSRE-375

The goal is to mimic frameworks like ShellCheck and Hadolint, and populate the AMO's wiki with detailed documentation about the static checks: https://github.com/mt-sre/addon-metadata-operator/wiki

It also adds a registry mechanism, so we only modify a single file to create a new validator.

## Upcoming work (future PR)
- Add `GetWikiURL()` on Validator struct
- Add helpers on Validator struct `Fail(), Succeed(), Error()` - that return standardized error/failure msgs /w link to Wiki (only `Fail()`) so tenants and users of this framework have access to more detailed documentation

### CODE REVIEW

**Comment by sugarraysam** [comment]
[HOLD] Needs to be rebased, branched off from #35

**Comment by yashvardhan-kukreja** [comment]
@sugarraysam #35 is merged. Can you rebase this PR?

**Comment by sugarraysam** [comment]
@yashvardhan-kukreja rebased, thanks !

File: pkg/validators/registry.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+}
+
+type registry struct {
+	*sync.Mutex
+	Data map[string]utils.Validator
+}
+
+// Add - update the registry in a thread-safe way. Called in init() functions
+func (r registry) Add(v utils.Validator) {
```
Comment (yashvardhan-kukreja) [inline pkg/validators/registry.go:26]:
```suggestion
func (r *registry) Add(v utils.Validator) {
```

**Review by yashvardhan-kukreja — APPROVED** [review]
lgtm!

**Pull request merged**